src/nvim/path.c

```diff
@@ -745,11 +743,11 @@ static bool is_unique(char_u *maybe_unique, garray_T *gap, int i)
     if (j == i)
       continue;        /* don't compare it with itself */
 
-    candidate_len = (int)STRLEN(maybe_unique);
-    other_path_len = (int)STRLEN(other_paths[j]);
-    if (other_path_len < candidate_len)
-      continue;        /* it's different when it's shorter */
-
+    candidate_len = STRLEN(maybe_unique);
+    other_path_len = STRLEN(other_paths[j]);
```

Comment:
You can declare the types here since the variables are only used inside the loop block.